evm/p2p/protocol.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import struct
+
+import rlp
+from rlp import sedes
+
+from evm.constants import NULL_BYTE
+
+
+class Command:
+    id = None
+    decode_strict = True
+    structure = []
+
+    @classmethod
+    def encode_payload(cls, data):
+        if isinstance(data, dict):  # convert dict to ordered list
+            if not isinstance(cls.structure, list):
+                raise ValueError("Command.structure must be a list when data is a dict")
+            data = [data[name] for (name, _) in cls.structure]
+        if isinstance(cls.structure, sedes.CountableList):
+            encoder = cls.structure
+        else:
+            assert len(data) == len(cls.structure)
```

Comment:
Should this raise an `rlp.EncodingError` instead?

Reply:
Yeah, definitely something other than an assert.